source/timed-out.js

```diff
@@ -32,10 +32,10 @@ module.exports = function (req, time) {
 		req.requestTimeoutTimer = setTimeout(() => {
 			clear();
 
-			if (req.connection.connecting) {
-				throwESOCKETTIMEDOUT();
-			} else {
+			if (req.connection.connecting || req.connection._connecting) {
```

Comment:
We don't need to check `._connecting` here. See: https://github.com/sindresorhus/got/blob/bd3315b6c61d20a68944831d8b3a05046d5554ad/index.js#L194